graphql_compiler/schema_transformation/merge_schemas.py

```diff
@@ -20,40 +42,58 @@
 )
 
 
-MergedSchemaDescriptor = namedtuple(
-    "MergedSchemaDescriptor",
-    (
-        "schema_ast",  # Document, AST representing the merged schema
-        "schema",  # GraphQLSchema, representing the same schema as schema_ast
-        "type_name_to_schema_id",
-        # Dict[str, str], mapping type name to the id of its schema, includes Interface, Object,
-        # Union, and Enum types
-    ),
-)
+@dataclass
```

Comment:
Consider adding `frozen=True` for each of these, since `frozen=False` is the default for dataclasses whereas the original `namedtuple` is immutable.